Dockerfile

```diff
@@ -67,6 +72,7 @@ RUN apt-get update && apt-get install -y \
 	xfsprogs \
 	libzfs-dev \
 	tar \
+	linux-image-`uname -r` \
```

Comment:
@southerngs I think the problem that is being conflated here is "official Docker builds" versus the main `Dockerfile` in the root of the repo which is meant for building Docker in one default context.  What is more noticeable in recent times is that this specific Dockerfile's isolation is now imperfect regarding total separation from the host. However, prior to dynamic linking of the Docker executable, and prior to lots of recent features, the encapsulation was pretty reasonable for a lion's share of cases.

Since that isn't the case anymore for producing (dynamically linked) binaries for a cross-section of supported distros, I think the Dockerfiles that people should be looking at in this discussion are found in the [docker/contrib/builder](https://github.com/docker/docker/tree/master/contrib/builder) subtree of this repo.

If you follow that repository path through its "**deb**" and "**rpm**" subtrees, you will find many subdirectories for variants and versions of popular Linux distributions, each with a specific `Dockerfile` that builds the Docker binary against the proper library versions/packages for that distro. Obviously in these cases there is no sense in which Docker can "isolate" you from the host OS because the goal of each of these Dockerfiles is to **create a runnable binary** for proper operation in that specific distro's environment which includes it's own variants of needed libraries and/or package names. This is obviously a slightly different use case than a standard use of Docker as simply a way to **run** binaries isolated from the host OS and filesystem.